spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/JobGroupUtils.java

```diff
@@ -55,10 +61,10 @@ public static <T> T withJobGroupInfo(
       SparkContext sparkContext, JobGroupInfo info, Supplier<T> supplier) {
     JobGroupInfo previousInfo = getJobGroupInfo(sparkContext);
     try {
-      setJobGroupInfo(sparkContext, info);
+      setJobGroupInfo(sparkContext, info, false);
       return supplier.get();
     } finally {
-      setJobGroupInfo(sparkContext, previousInfo);
+      setJobGroupInfo(sparkContext, previousInfo, true);
```

Comment:
why set overwrite to true here?

Reply:
> why set overwrite to true here?

use the previous jobGroupInfo to overwrite iceberg jobGroupInfo to avoid affecting subsequent jobs